pkg/appmgmt/appmgmt_recovery.go

```diff
@@ -63,14 +63,18 @@ func (svc *AppManagementService) recoverApps() (map[string]interfaces.ManagedApp
 				return pods[i].CreationTimestamp.Unix() < pods[j].CreationTimestamp.Unix()
 			})
 
+			// Track pods that we have already seen in order to
+			// skip redundant handling of async events in RecoveryDone
+			seenPods := make(map[string]bool)
 			for _, pod := range pods {
 				if utils.NeedRecovery(pod) {
 					app := svc.podEventHandler.HandleEvent(general.AddPod, general.Recovery, pod)
 					recoveringApps[app.GetApplicationID()] = app
 				}
+				seenPods[string(pod.UID)] = true
```

Comment:
This is updated outside of the "if" branch. Which might contain all kinds of pods not scheduled by Yunikorn.
We need to be inside the branch, plus, only store pods that are not running.

If a pod is running, you don't know whether we have a new event for it or not. You can't filter out events which are about an active pod.